src/utils/wallet.js

```diff
@@ -70,26 +74,68 @@ export function catchTransactionError(e: Object, type: string, tx: Object) {
 }
 
 // handle eth_signTransaction
-export async function signTransaction(trx: Object, wallet: Object): Promise<string> {
+export function signTransaction(trx: Object, wallet: Object): Promise<string> {
   wallet.provider = providers.getDefaultProvider(NETWORK_PROVIDER);
   if (trx && trx.from) {
     delete trx.from;
   }
-  const result = await wallet.sign(trx);
-  return result;
+  return wallet.sign(trx);
 }
 
 // handle eth_sign
-export async function signMessage(message: any, wallet: Object): Promise<string> {
+export function signMessage(message: any, wallet: Object): string {
   wallet.provider = providers.getDefaultProvider(NETWORK_PROVIDER);
   // TODO: this method needs to be replaced when ethers.js is migrated to v4.0
-  const result = ethSign(message, wallet.privateKey);
-  return result;
+  return ethSign(message, wallet.privateKey);
 }
 
 // handle personal_sign
-export async function signPersonalMessage(message: string, wallet: Object): Promise<string> {
+export function signPersonalMessage(message: string, wallet: Object): Promise<string> {
   wallet.provider = providers.getDefaultProvider(NETWORK_PROVIDER);
-  const result = await wallet.signMessage(isHexString(message) ? ethers.utils.arrayify(message) : message);
-  return result;
+  return wallet.signMessage(isHexString(message) ? ethers.utils.arrayify(message) : message);
+}
+
+// we use basic AsyncStorage implementation just to prevent backup being stored in same manner
+export async function getWalletFromStorage(dispatch: Dispatch, appSettings: Object) {
```

Comment:
Small suggestions: I'd avoid the `get` on function names and use something like `walletFromStorage` or just `storedWallet`, for example. Also, what do you think about adding a type for what this function returns? I'm not sure if it would be worth it or not :thinking:

Reply:
Why would you avoid the `get` on function names?

Reply:
I think `get` adds no information to the name whatsoever. All of the functions that return information will get it from somewhere, we don't need to add that `get` into the name to make it explicit. Unless there's a constraint, I'd avoid things like `get`, `calculate`, `check`, `do`, etc.

Reply:
As for me, those addons make code more readable :)

Reply:
Alright, that's fine :+1:

Reply:
👍